services/openvpn/service/factory.go

```diff
@@ -112,21 +112,16 @@ type OpenvpnConfigNegotiator struct {
 	vpnConfig openvpn_service.VPNConfig
 }
 
-// ConsumeConfig doesn't do anything on the openvpn side, since it's not required here
-func (ocn *OpenvpnConfigNegotiator) ConsumeConfig(json.RawMessage) error {
-	return nil
-}
-
 // ProvideConfig returns the config for user
-func (ocn *OpenvpnConfigNegotiator) ProvideConfig() (session.ServiceConfiguration, error) {
+func (ocn *OpenvpnConfigNegotiator) ProvideConfig(json.RawMessage) (session.ServiceConfiguration, error) {
 	return &ocn.vpnConfig, nil
```

Comment:
I don't see anywhere where `ServiceConfiguration` interface implementation would return any kind of error. Not sure if its needed, but we could ALWAYS return a valid 'pseudo' configuration without possibility of error.

Reply:
Maybe error is needed to signal underlying caller that we cannot provide config - therefore drop consumer session or whatever

Reply:
Should we cover it in this PR?

Reply:
Isn't it's working this way already? My comment was more an answer to @zolia

Reply:
yep, correct, it can fail if we cannot provide config.